brig/cmd/brig/commands/check.go

```diff
@@ -0,0 +1,121 @@
+package commands
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	apps_v1 "k8s.io/api/apps/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	Root.AddCommand(check)
+}
+
+const (
+	checkUsage = `Checks the status of your Brigade installation
+
+Specifically, it checks if the Controller, API Server and Kashti deployments have more than one available replicas (pods).
+`
+)
+
+var check = &cobra.Command{
+	Use:   "check",
+	Short: "Checks the status of your Brigade installation",
+	Long:  checkUsage,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return checkBrigadeSystem()
+	},
+}
+
+func checkBrigadeSystem() error {
+	/*
+		If you run `kubectl get deploy --show-labels` on the namespace where Brigade is installed, you'll see
+		NAME                                DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE    LABELS
+		mybrigade-brigade-api               1         1         1            1           170m   app.kubernetes.io/instance=mybrigade,app.kubernetes.io/managed-by=Tiller,app.kubernetes.io/name=mybrigade-brigade-api,helm.sh/chart=brigade-1.0.0,role=api
+		mybrigade-brigade-ctrl              1         1         1            1           170m   app.kubernetes.io/instance=mybrigade,app.kubernetes.io/managed-by=Tiller,app.kubernetes.io/name=mybrigade-brigade-ctrl,helm.sh/chart=brigade-1.0.0,role=controller
+		mybrigade-brigade-generic-gateway   1         1         1            1           170m   app.kubernetes.io/instance=mybrigade,app.kubernetes.io/managed-by=Tiller,app.kubernetes.io/name=mybrigade-brigade,helm.sh/chart=brigade-1.0.0,role=gateway,type=generic
+		mybrigade-kashti                    1         1         1            1           170m   app=kashti,chart=kashti-0.1.1,heritage=Tiller,release=mybrigade
+	*/
+
+	c, err := kubeClient()
+	if err != nil {
+		return err
+	}
+
+	// check deployments
+	deployList, err := c.AppsV1().Deployments(globalNamespace).List(meta_v1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	apiDeploymentFound := false
+	controllerDeploymentFound := false
+	kashtiDeploymentFound := false
+
+	for _, deployment := range deployList.Items {
+		if lbl := deployment.Labels["app.kubernetes.io/name"]; lbl != "" {
+			if strings.HasSuffix(lbl, "-brigade-api") && deployment.Labels["role"] == "api" {
+				apiDeploymentFound = true
+				reportDeployStatus(deployment, "Brigade API Server")
```

Comment:
One question -- do we want to error out if Kashti or the API are unavailable?
As they are not essential, this might be informational -- thoughts?

Reply:
yeah, you're right. I also altered the Deploy message to make it more consistent with `kubectl get deploy`, let me know what you think.